### PR TITLE
Create egovframe-ide.rb

### PR DESCRIPTION
**Pull request by overpassion**
add eGovFrame IDE 4.1.0

eGovFrame, the e-Government Standard Framework,
is a platform-specific standardized development framework for public sector IT projects in Korea.
It is developed by the government of the Republic of Korea and
can be used by every person all over the world.

* Homepage
https://www.egovframe.go.kr/eng/main.do

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/e/egovframe-ide.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+cask "egovframe-ide" do
+  version "4.1.0"
+
+  app_name = ""
```
Comment (SMillerDev) [inline Casks/e/egovframe-ide.rb:4]:
```suggestion
  arch arm: "AArch64", intel: "x86_64"
```

File: Casks/e/egovframe-ide.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  desc "Open-source framework by South Korea for web-based public service development"
+  homepage "https://www.egovframe.go.kr/"
+
+  livecheck do
+    url "https://www.egovframe.go.kr/home/sub.do?menuNo=39"
+    regex(/개발환경 (\d+\.\d+\.\d+) 플러그인/i)
+    strategy :page_match
+  end
+
+  app app_name
```
Comment (SMillerDev) [inline Casks/e/egovframe-ide.rb:29]:
```suggestion
  app "eGovFrameDev-#{version}-Mac-#{arch}.app"
```

File: Casks/e/egovframe-ide.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    url "https://maven.egovframe.go.kr/publist/HDD1/public/eGovFrameDev-#{version}-Mac-AArch64.dmg"
+    app_name = "eGovFrameDev-#{version}-Mac-AArch64.app"
+  end
+  on_intel do
+    sha256 "b3c652165914f7ccc05eae9e4730954a0a2cf0d361331a14f2c291fa51510652"
+
+    url "https://maven.egovframe.go.kr/publist/HDD1/public/eGovFrameDev-#{version}-Mac-x86_64.dmg"
+    app_name = "eGovFrameDev-#{version}-Mac-x86_64.app"
+  end
```
Comment (SMillerDev) [inline Casks/e/egovframe-ide.rb:17]:
```suggestion
  url "https://maven.egovframe.go.kr/publist/HDD1/public/eGovFrameDev-#{version}-Mac-#{arch}.dmg"
  sha256 arm: "7e7ece1b05a6e4ccfca522a44f4a4e0230a164f8542df0e78c0181efc039590e",
         intel: "b3c652165914f7ccc05eae9e4730954a0a2cf0d361331a14f2c291fa51510652"
```

File: Casks/e/egovframe-ide.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  livecheck do
+    url "https://www.egovframe.go.kr/home/sub.do?menuNo=39"
+    regex(/개발환경 (\d+\.\d+\.\d+) 플러그인/i)
+    strategy :page_match
+  end
+
+  app app_name
+
+  postflight do
+    set_ownership "/Applications/#{app_name}"
```
Comment (SMillerDev) [inline Casks/e/egovframe-ide.rb:32]:
```suggestion
    set_ownership "/Applications/eGovFrameDev-#{version}-Mac-#{arch}.app"
```

Comment (overpassion) [reply Casks/e/egovframe-ide.rb]: Thanks for the suggestion code.

File: Casks/e/egovframe-ide.rb (hunk 3 above)
Comment (overpassion) [reply Casks/e/egovframe-ide.rb]: Thanks for the suggestion code. Applied.

File: Casks/e/egovframe-ide.rb (hunk 2 above)
Comment (overpassion) [reply Casks/e/egovframe-ide.rb]: Thanks for the suggestion code. Applied.

File: Casks/e/egovframe-ide.rb (hunk 1 above)
Comment (overpassion) [reply Casks/e/egovframe-ide.rb]: Thanks for the suggestion code. Applied.

File: Casks/e/egovframe-ide.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    url "https://www.egovframe.go.kr/home/sub.do?menuNo=39"
+    regex(/개발환경 (\d+\.\d+\.\d+) 플러그인/i)
+    strategy :page_match
+  end
+
+  app "eGovFrameDev-#{version}-Mac-#{arch}.app"
+
+  postflight do
+    set_ownership "/Applications/eGovFrameDev-#{version}-Mac-#{arch}.app"
+  end
```
Comment (bevanjkay) [inline Casks/e/egovframe-ide.rb:23]: Is there a specific reason this is required?

Comment (overpassion) [reply Casks/e/egovframe-ide.rb:23]: I think it's unnecessary, so I'll delete it. thank you

**Comment by miccal** [comment]
Thank you @overpassion.

**Review by miccal — APPROVED** [review]

**Pull request merged**